tests/ConfigControllerTest.php

```diff
@@ -19,31 +34,137 @@ public function testConfigurationIsSavedWithEmptyValues()
+        $mockedScriptsJsonResponse = $this->getMockResponse('json_response/v3_content_scripts.json');
+
+        $mock = new MockHandler([
+            // REST-API call: stores/<store_hash>/v2/store
+            new Response(200, ['Content-Type' => 'application/xml'], $mockedStoresXmlResponse),
+            // REST-API call: stores/<store_hash>/v3/content/scripts
+            new Response(200, ['Content-Type' => 'application/json'], $mockedScriptsJsonResponse),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $httpClientMock = new Client(['handler' => $handler]);
+        $configController = new ConfigController($httpClientMock);
+
+        $request = Request::create('/config', 'POST', [
             'store_hash' => 'test123',
             'access_token' => 'testAcessToken123',
             'context' => 'stores/test123',
             'shopkey' => '123test',
             'active_status' => 'on'
         ]);
-        $configController = new ConfigController();
         $configController->handleConfiguration($request);
 
-        $store = Store::where('domain', 'stores/test123')->first();
+        /** @var Store $store */
+        $store = Store::where('context', 'stores/test123')->first();
         $config = Config::where('store_id', $store['id'])->first();
 
         $this->assertTrue(boolval($config['active']));
         $this->assertEquals($store['id'], (int) $config['store_id']);
         $this->assertSame('123test', $config['shopkey']);
+
+        $this->assertEquals(1, $store->scripts->count());
+        $script = $store->scripts()->first();
+
+        $this->assertEquals('Findologic', $script->name);
+        $this->assertEquals('9dd04fae-d45d-5a64-a45d-62d14d2c62b5', $script->uuid);
+        $this->assertEquals(str_replace('stores/', '', $store->context), $script->store_hash);
+        $this->assertEquals($store->id, $script->store_id);
+    }
+
+    public function testConfigurationDoesNotSaveScriptWhenActiveIsNotSet()
+    {
+        $mockedStoresXmlResponse = $this->getMockResponse('/xml_response/v2_stores.xml');
+
+        $mock = new MockHandler([
+            new Response(200, ['Content-Type' => 'application/xml'], $mockedStoresXmlResponse),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $httpClientMock = new Client(['handler' => $handler]);
+        $configController = new ConfigController($httpClientMock);
+
+        $request = Request::create('/config', 'POST', [
+            'store_hash' => 'test123',
+            'access_token' => 'testAcessToken123',
+            'context' => 'stores/test123',
+            'shopkey' => '123test',
+            'active_status' => null
+        ]);
+        $configController->handleConfiguration($request);
+
+        /** @var Store $store */
+        $store = Store::where('context', 'stores/test123')->first();
+        $config = Config::where('store_id', $store['id'])->first();
+
+        $this->assertFalse(boolval($config['active']));
```

Comment:
Why does this need to be cased to an boolean? Shouldn't `active` be a bool already?

Reply:
Works locally, fails on Github
![Screen Shot 2021-02-03 at 13 47 05](https://user-images.githubusercontent.com/511559/106749074-5cfff200-6626-11eb-9628-99a14451110f.png)

Reply:
> SQLite does not have a separate Boolean storage class. Instead, Boolean values are stored as integers 0 (false) and 1 (true).

This is why, is `bool_val` fine in this case or do you have any other suggestions?

Reply:
But how can this work locally then? Aren't you using SQLite there as well?

Reply:
Seems not since db connection is defined in `env.ci`

Reply:
Okay, but instead of converting this in the test I would prefer that this is done in the actual sources. Casting a boolean to a boolean won't change anything, but casting 0/1 to a boolean, allows for proper usage. If you were using SQLite in the background, the application would not properly work, as it would literally use 0/1 as integers.

Reply:
https://github.com/findologic/plugin-bigcommerce-di/pull/12/commits/f3c7fe60f230d0386cb101c66ec3eeced371de35